packages/medusa/src/api/store/carts/[id]/complete/route.ts

```diff
@@ -1,22 +1,48 @@
 import { completeCartWorkflow } from "@medusajs/core-flows"
-import { MedusaError } from "@medusajs/utils"
+import {
+  MedusaError,
+  ModuleRegistrationName,
+  TransactionState,
+} from "@medusajs/utils"
 import { MedusaRequest, MedusaResponse } from "../../../../../types/routing"
+import { prepareRetrieveQuery } from "../../../../../utils/get-query-config"
+import { refetchEntity } from "../../../../utils/refetch-entity"
 import { refetchOrder } from "../../../orders/helpers"
 import { refetchCart } from "../../helpers"
 import { defaultStoreCartFields } from "../../query-config"
 import { StoreCompleteCartType } from "../../validators"
-import { prepareRetrieveQuery } from "../../../../../utils/get-query-config"
 
 export const POST = async (
   req: MedusaRequest<StoreCompleteCartType>,
   res: MedusaResponse
 ) => {
   const cart_id = req.params.id
 
+  const eventBusModule = req.scope.resolve(ModuleRegistrationName.EVENT_BUS)
+
   const { errors, result } = await completeCartWorkflow(req.scope).run({
     input: { id: cart_id },
     context: { transactionId: cart_id },
     throwOnError: false,
+    events: {
+      onFinish: async ({ transaction }) => {
+        const state = transaction.getState()
+
+        // We only want to emit the order.placed event if the workflow completed
+        if (state !== TransactionState.DONE) {
+          return
+        }
+
+        const result = await refetchEntity(
+          "order_cart",
+          { cart_id },
+          req.scope,
+          ["order_id", "cart_id"]
+        )
+
+        await eventBusModule.emit("order.placed", { id: result.order_id })
```

Comment:
nit: I would just create an events.ts file in the utils package under the order dir with the following
```ts
const eventBaseNames: [
  "order"
] = [
  "order",
]

export const OrderEvents = {
  ...buildEventNamesFromEntityName(
    eventBaseNames,
    Modules.ORDER
  ),
  'order_placed': 'order.placed'
}
```
and then consume it here, not necessarely here but just a thoughts

Reply:
Yeah think this is a good idea!